File: Medium/GroupAnagrams/GroupAnagrams.cpp
// https://leetcode.com/problems/group-anagrams/

class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        vector<vector<string>> result;
        unordered_map<string, vector<string>> wordMap;
        for(int i = 0; i < strs.size(); i++){
            string tempWord = strs[i];
            std::sort(tempWord.begin(), tempWord.end());
            wordMap[tempWord].push_back(strs[i]);
        }

        for (auto it = wordMap.begin(); it != wordMap.end(); it++){
            result.push_back(it->second);
        }
        return result;
    }

};

File: Medium/GroupAnagrams/GroupAnagrams.py
# https://leetcode.com/problems/group-anagrams/

class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        # groups = defaultdict(list)
        # for word in strs:
        #     groups[''.join(sorted(word))].append(word)
        # return groups.values()







        groups = defaultdict(list)
        for word in strs:
            count = [0 for _ in range(26)]
            for ch in word:
                count[ord(ch)-ord('a')]+=1
            groups[tuple(count)].append(word)
        return groups.values()
        